dbcon/joblist/tuple-bps.cpp

```diff
@@ -358,7 +357,8 @@ TupleBPS::TupleBPS(const pColScanStep& rhs, const JobInfo& jobInfo) :
 
     initExtentMarkers();
     fQtc.stepParms().stepType = StepTeleStats::T_BPS;
-
+    fJoinMemLimit.reset(new int64_t(*jobInfo.umMemLimit == 0 ? fRm->getConfiguredUMMemLimit() / 4 : min((uint64_t)*jobInfo.umMemLimit, fRm->getConfiguredUMMemLimit() / 4)));
```

Comment:
We do, because the default for calls to getMemory use a shared_ptr. We can probably do a reset() rather than all that size figuring. Since coding this, I've added the ability to accept a NULL shared_ptr in getMemory().